fixitfelix/fix.py

```diff
@@ -151,3 +152,85 @@ def export_correct_data(
                     write_chunks_to_file(
                         tdms_writer, index_ranges, group, channel
                     )
+
+
+def export_correct_data(
+    filename: str, meta: source.MetaData, output_file: str
+) -> None:
+    """ Accepts either a path to a tdms file or to a folder with just tdms files to correct.
+    The name of the resulting folder or file is defined by output_file.
+    If output_file is empty, the name of the resulting file or folder is the previous name with '_corrected' as suffix.
+    All files in a folder will retain their previous name with '_corrected' suffix.
+    In case of a folder as input all tdms files are first checked for consistency so no file is corrected before each file is checked.
+    Afterwards all files are corrected and exported. This prevents cases where a later file is not valid for correction.
+    If a single file is given, the file is checked and corrected immediately.
+
+    Arguments:
+    filename: Path to the tdms file or folder with tdms files to correct.
+    meta: MetaData dict that contains all information needed for correction.
+    output_file: File path for the corrected TDMS file or folder.
+    """
+
+
+    #Determines generalized export path
+
+    path = pathlib.Path(filename)
+
+    p = either.Right(path) | error_handling.check_input_path
+    if isinstance(p, either.Left):
+            raise Exception(error_handling.ERROR_DESCRIPTIONS.get(p._value))
+
+    if output_file == "":
+        name = path.with_suffix("").name + "_corrected"
+        export_path = path.parent.joinpath(name)
+    else:
+        export_path = pathlib.Path(output_file)
+
+    p = either.Right(export_path) | check_export_path
+    if isinstance(p, either.Left):
+        raise Exception(error_handling.ERROR_DESCRIPTIONS.get(p._value))
+
+
+    # Directory and single file are handled seperately
+
+    if path.is_dir():
+        p = either.Right(path) | error_handling.check_dir_empty
+        if isinstance(p, either.Left):
+            raise Exception(error_handling.ERROR_DESCRIPTIONS.get(p._value))
+
+        if export_path.exists():
+            shutil.rmtree(export_path)
+        os.mkdir(export_path)
+
+
+        # Checks each file in folder for consistency
+
+        files_in_dir = len([f for f in path.iterdir()])
+        file_count = 0
+        pre_files = []
+        for tdms_file in path.iterdir():
+            pre_data = {}
```

Comment:
We should add an explanation of the new feature to the README file no clear out what the requirements to the input and output folders are.

Reply:
Definitely. The input folder has to include at least one tdms file and no other non-tdms files. Like mentioned above a TDMSPATH_NONEXISTENT error is returned in case of a non-tdms file in the input folder. Before each preprocessing of a file in the folder, the file name is mentioned, so you can easily figure out which file causes the error.